src/client/src/components/patientmodal/Modal.js

```diff
@@ -1,20 +1,238 @@
-import './Modal.css';
-
-export const Modal = ({ patient_info, handleClose, show,}) => {
-  const showHideClassName = show ? "modal display-block" : "modal display-none";
-  return (
-    <div className={showHideClassName}>
-      <section className="modal-main">
-        <div className="modal_content">
-        <div className="patient-field title"> {patient_info.name} </div>
-        <div className="patient-field">id: {patient_info.id} </div>
-        <div className="patient-field">gender: {patient_info.gender} </div>
-        <div className="patient-field">birthdate: {patient_info.birthdate} </div>
-        <button className="button" type="button" onClick={handleClose}>
-          Close
-        </button>
+import React from 'react';
+import Select from 'react-select';
+import './Modal.scss';
+import './table.scss';
+import GROUP_MAPS from'./group_maps';
+import Table from 'rc-table';
+import { jsPDF } from "jspdf";
+import { CSVLink, CSVDownload } from "react-csv";
+
+export class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    const groups = this.getGroups(this.props.patient_data);
+    const groupOptions = this.getGroupOptions(groups);
+    this.state = {
+      groups: groups,
+      options: groupOptions,
+      selectedGroup: groupOptions[0],
+      columns: [],
+      rows: []
+    };
+    
+  }
+
+  getPath = (obj, path = "") => {
+    return path.split(".").reduce((out, key) => out ? out[key] : undefined, obj)
+  }
+
+  handleDownload = ()=>{
+    const doc = new jsPDF();
+    const id = this.state.groups['Patient'][0]['resource']['id']
+    const last_name = this.state.groups['Patient'][0]['resource']['name'][0]['family']
+    const first_name = this.state.groups['Patient'][0]['resource']['name'][0]['given'][0]
+    doc.text("GROUP" + this.state.selectedGroup.value, 20, 20);
+    doc.text(JSON.stringify(this.state.columns), 20, 30);
+    var cur_row = 40
+    this.state.rows.forEach(row=>{
+      doc.text(JSON.stringify(row), 20, cur_row);
+      cur_row += 10;
+    })
+    doc.save(first_name + ' ' + last_name + ' ' + this.state.selectedGroup + ".pdf");
+  }
```

Comment:
The download needs to include all columns of all entries, not just the ones we display. It should also download everything, not just the category that's selected.

Reply:
Ya I'll work on that for the csv, this function(handle download)  is outdated tho, and i forgot to delete it. It was for the pdf version which was not working very well. I'll remove it next push